src/components/ClassSearch.vue

```diff
@@ -209,6 +209,32 @@ class BooleanFilter extends Filter {
   }
 }
 
+class ArrayFilter extends Filter {
+  constructor (name, shortName, attributeNames, SubfilterType, subfilterArguments, mode) {
+    var subfilter = new SubfilterType(name, shortName, attributeNames, ...subfilterArguments, mode);
+    var comparator = subfilter.filter;
+    super(name, shortName, comparator, attributeNames, mode);
+    this.subfilter = subfilter;
+  }
+  matches (subject) {
+    // starting value of true for and, false for or
+    var isMatch = !this.combine(true, false);
```

Comment:
I am not sure that searching by instructor is worth the slowdown (probably caused by all these loops).  We should think about ways to improve the speed of this.  One thought i had is joining the instructors with commas into a single string when we pull the course catalog, then searching using a regex that avoids the comma boundaries.